### DISCUSSION TITLE
Support for Android resources in MAUI libraries (e.g /layout, /drawable, /values, etc)?

### DISCUSSION THREAD

**Discussion by nCastle1** [dotnet/maui#9824] [Q&A]
I'm trying to port a custom control from Xamarin.Forms to MAUI. The control was wrapping Xamarin.Android and Xamarin.iOS implementations of the control.

Porting the control to .NET 6+MAUI is going ok, but I can't get anything related to Android resources to work. Even following the same exact pattern as a MAUI project, Visual Studio isn't recognizing xml files in `Platforms/Android/Resources/*` as `AndroidResource`, and I can't manually change that.

I might just rewrite the layout in code, but that doesn't work as well for things like drawables.

**Comment by xtuzy** [comment]
maybe you can create a android class library to store android resources

**Comment by bulubuloa** [comment]
@nCastle1 Have you dealt with this? I have an issue accessing the Android resource XML via ID.
It seems MAUI does not generate resource file like Resource.designer.cs

**Reply by Pastajello** [reply]
So im not the only one. It should be possible with <Namespace>.Resources  BUUUUT I have not been able to make it work at all. This project have it workig so i dunno: 
[![image](https://github.com/dotnet/maui/assets/28345098/76372f34-2055-4261-b7e9-e67db20bb353)](https://github.com/ericbrunner/TaskyApp.Maui.SingleProject)


![image](https://github.com/dotnet/maui/assets/28345098/1be41cee-d5db-40bf-8787-ac1594cd71ba)

**Reply by Pastajello** [reply]
Fun fact, on VS 2022 preview it works nicely. I add a drawable to the maui project, its there, sitting in the Resource class . Vs for Mac? Hardly.

![obraz](https://github.com/dotnet/maui/assets/28345098/f8488b21-4750-467e-b74e-4eb2d078b6ba)

**Comment by chrismcna** [comment]
In my library I added a folder Platforms/Android/Resources/drawable 
then copied to the png into there, set it as "Embedded resource", the image then appeared in Resource.Drawable, code complies and icon displayed on notification in android that generated by the maui library

visual studio 2026 insider on .net10
